x/tx/signing/textual/any.go

```diff
@@ -113,3 +130,19 @@ func (ar anyValueRenderer) Parse(ctx context.Context, screens []Screen) (protore
 
 	return protoreflect.ValueOfMessage(anyMsg.ProtoReflect()), nil
 }
+
+// toAny converts the proto Message to a anypb.Any.
+// The input msg can be:
+// - either a anypb.Any already (in which case there's nothing to do),
+// - or a dynamicpb.Message.
+func toAny(msg proto.Message) (*anypb.Any, error) {
```

Comment:
Here and below for similar converters - I'm quite confused by this. Don't we know that `Any`, `Duration`, etc, are in fact registered?

Reply:
The fact that we import `google.golang.org/protobuf/types/known/{any,duration...}pb` indeed guarantees that `Any`, `Duration` are registered.

However, when they are nested in another message that's not registered, then they are dynamically created using dynamicpb. Consider a custom Msg, suppose it's only registered in gogo, and not in protov2:

```
message MsgCustom {
  google.protobuf.Any a = 1;
}
```

Then when Textual receives this message, it sees `MsgCustom` and will use dynamicpb to generate the whole proto.Message, including the nested field.

Reply:
Thanks for the explanation. It would be nice if we didn't have to complicate the textual code for this, but I don't know enough to second-guess the choices you've made here.